pynicotine/logfacility.py

```diff
@@ -220,6 +222,38 @@ def update_folder_paths(self):
     def open_log(self, folder_path, basename):
         self._log_file_operation(folder_path, basename, self.open_log_callback)
 
+    def _get_log_lines(self, file_handle, num_lines):
+
+        file_handle.seek(0, os.SEEK_END)
+
+        blocks = deque()
+        block_number = 1
+        file_size = file_handle.tell()
+        lines_left = num_lines + 1
+
+        while lines_left > 0:
+            read_offset = file_size - (block_number * self.READ_BLOCK_SIZE)
```

Comment:
`log_file.read_offset -= (block_number * self.READ_BLOCK_SIZE)`

Reply:
Got rid of the block number entirely. I used it while experimenting, but it became redundant eventually.